invenio_previewer/extensions/video_videojs.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 Northwestern University.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Render video file using videojs."""
+
+from flask import current_app, render_template
+
+from ..proxies import current_previewer
+from ..utils import dotted_exts
+
+# The video file world is complex, because support depends on
+# hardware, OS, software and browser. A reasonable choice of supported
+# extensions (https://en.wikipedia.org/wiki/HTML5_video#Browser_support) that
+# are least likely to lead to an unsupported warning for end-users are
+# mp4 and webm files. However, this is made customizable per instance.
+previewable_extensions = current_app.config.get("PREVIEWER_VIDEO_EXTS", ["mp4", "webm"])
+
+
+def can_preview(file):
+    """Determine if the given file can be previewed."""
+    return file.has_extensions(*dotted_exts(previewable_extensions))
+
```

Comment:
Is there any size limitation to the file that can be previewed?

Reply:
Should there be a sanitization process implemented before previewing the file?

Reply:
> Is there any size limitation to the file that can be previewed?

There probably should be and it should be a configuration like there is for [XML files for instance](https://github.com/inveniosoftware/invenio-previewer/blob/master/invenio_previewer/extensions/xml_prismjs.py#L39) (with a different config IMHO since videos are expected to be larger). We are testing this internally in our staging environment to see what those infrastructure configurations could look like. 

> sanitization process implemented before previewing the file?

I didn't include the ogg/theora video container format/codec as default for instance, since it has shown to be a potential vector for attacks (mostly because it's not really in development anymore). The other codecs are well maintained, and I feel comfortable with their security level. I can't think of other processing we can really do without becoming a full-blown media server ([CDS doesn't do that either](https://github.com/CERNDocumentServer/cds-videos/blob/main/cds/modules/previewer/extensions/video.py) ...).  

Something else that could be added in the future is whether the file is local or not. We only use local files, so I can't contribute in that direction based on realistic experiences though.